woocommerce/woocommerce
# Display notices properly in Marketplace page

### All Submissions:


### Changes proposed in this Pull Request:

WooCommerce Admin hides some notices on WooCommerce admin pages, like the WP update notice. But other WP notices (like the "regenerating thumbnails" notice) may still appear. This PR positions them in a suitable place on the Marketplace page.

* Centre the content on the Marketplace page, in line with the design.
* Add a `.wp-header-end` div to the page, where WP will render `admin_notices` notices.
* Style notices to look OK on Marketplace and My Subscriptions pages.

Closes 11000-gh-Automattic/woocommerce.com.

### How to test the changes in this Pull Request:

1. Check out this branch and build assets.
2. View the Marketplace page (example http://localhost:8100/wp-admin/admin.php?page=wc-addons). On a large screen, you should see that the content is centred.
3. View the My Subscriptions page (example http://localhost:8100/wp-admin/admin.php?page=wc-addons&section=helper). The content on the page should appear as normal. (The breadcrumb will be changed from `Extensions` in a WooCommerce Admin [PR](https://github.com/woocommerce/woocommerce-admin/pull/7471).)
4. Add this temporary code in `includes/admin/class-wc-admin.php:43` to display notices on the Marketplace page:

```php
add_action( 'admin_notices', function () {
    echo '<div class="notice notice-warning is-dismissible">
        <p>This notice appears on the settings page.</p>
    </div>
    <div class="notice notice-error is-dismissible">
        <p>This notice appears on the settings page.</p>
    </div>';
} );
```

5. View the Marketplace and My Subscriptions pages again. The notices should look OK, and when you dismiss them, the pages should still look OK.

### Other information:




### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### FOR PR REVIEWER ONLY:


### Screenshots

<img width="500" src="https://user-images.githubusercontent.com/1647564/129073818-5ae22260-1d82-4cb7-9a56-6dcd4945cc62.png">
